mysql-test/t/bug84640.test

```diff
@@ -0,0 +1,142 @@
+#
+# Bug #25451091 / #84640 "symlinks in CREATE TABLE ... DATA DIRECTORY"
+# lp:1654256 "MyISAM CREATE TABLE DATA DIRECTORY check race"
+#
+
+--source include/have_debug_sync.inc
+--source include/have_symlink.inc
+--source include/count_sessions.inc
+
+--echo #
+--echo # Tests for MyISAM handling of symlinks
+--echo #
+
+--let $datadir= `SELECT @@datadir`
+--let $tmpdir= `SELECT @@tmpdir`
+--let $custom_data_directory= $tmpdir/bug84640
+
+--echo #
+--echo # Bug 1654256: symlinks in CREATE TABLE ... DATA DIRECTORY, last path component
+--echo #
+
+--mkdir $custom_data_directory
+
+--replace_result $custom_data_directory custom_data_directory
+eval CREATE TABLE t1 (a INT) ENGINE=MyISAM DATA DIRECTORY='$custom_data_directory';
+
+INSERT INTO t1 VALUES (1), (2);
+
+FLUSH TABLES;
+
+--connect(con1,localhost,root,,)
+
+SET DEBUG_SYNC="before_opening_datafile SIGNAL open_ready WAIT_FOR finish_open";
+
+send SELECT * FROM t1;
+
+--connection default
+
+SET DEBUG_SYNC="now WAIT_FOR open_ready";
+
+--remove_file $custom_data_directory/t1.MYD
+--exec ln -s $datadir/mysql/user.MYD $custom_data_directory/t1.MYD
+
+SET DEBUG_SYNC="now SIGNAL finish_open";
+
+--connection con1
+
+--replace_result $custom_data_directory custom_data_directory
+--error 29
```

Comment:
Yes, I noticed this as well, but unfortunately there is no symbolic names for codes with values less than 1000.